File: StudentOglasi/UI/studentoglasi_mobile/lib/models/PrijavaStipendija/prijave_stipendija.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:studentoglasi_mobile/models/PrijavaDokumenti/prijava_dokumenti.dart';

import '../StatusPrijave/statusprijave.dart';
import '../Stipendija/stipendija.dart';
import '../Student/student.dart';

part 'prijave_stipendija.g.dart';

@JsonSerializable()
class PrijaveStipendija {
  int? studentId;
  int? stipendijaId;
  String? dokumentacija;
  String? cv;
  double? prosjekOcjena;
  StatusPrijave? status;
  Stipendije? stipendija;
  Student? student;
  DateTime? vrijemePrijave;
  List<PrijavaDokumenti>? dokumenti;

  PrijaveStipendija(
      this.studentId,
      this.stipendijaId,
      this.dokumentacija,
      this.cv,
      this.prosjekOcjena,
      this.status,
      this.stipendija,
      this.student,
      this.vrijemePrijave,
      this.dokumenti);
  factory PrijaveStipendija.fromJson(Map<String, dynamic> json) =>
      _$PrijaveStipendijaFromJson(json);
  Map<String, dynamic> toJson() => _$PrijaveStipendijaToJson(this);
}

File: StudentOglasi/StudentOglasi.Services/Database/PrijaveStipendija.cs
using System;
using System.Collections.Generic;

namespace StudentOglasi.Services.Database;

public partial class PrijaveStipendija
{
    public int Id { get; set; }
    public int StudentId { get; set; }

    public int StipendijaID { get; set; }

    public virtual ICollection<PrijavaDokumenti> Dokumenti { get; set; } = new List<PrijavaDokumenti>();

    public string? Cv { get; set; }

    public decimal? ProsjekOcjena { get; set; }

    public DateTime? VrijemePrijave { get; set; }

    public int StatusId { get; set; }

    public virtual StatusPrijave Status { get; set; } = null!;

    public virtual Stipendije Stipendija { get; set; } = null!;

    public virtual Studenti Student { get; set; } = null!;
}
